cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ArtifactHttpHandler.java

```diff
@@ -217,6 +218,29 @@ public void getArtifactInfo(HttpRequest request, HttpResponder responder,
   }
 
   @GET
+  @Path("/namespaces/{namespace-id}/artifacts/{artifact-name}/versions/{artifact-version}/parents")
+  public void getArtifactParents(HttpRequest request, HttpResponder responder,
+                                 @PathParam("namespace-id") String namespaceId,
+                                 @PathParam("artifact-name") String artifactName,
+                                 @PathParam("artifact-version") String artifactVersion,
+                                 @QueryParam("scope") @DefaultValue("user") String scope)
+    throws Exception {
+
+    NamespaceId namespace = validateAndGetScopedNamespace(Ids.namespace(namespaceId), scope);
+    Id.Artifact artifactId = validateAndGetArtifactId(namespace, artifactName, artifactVersion);
+
+    try {
+      ArtifactDetail detail = artifactRepository.getArtifact(artifactId);
+      ArtifactParentWrapper wrapper = new ArtifactParentWrapper(detail.getMeta().getUsableBy());
+      responder.sendJson(HttpResponseStatus.OK, wrapper, ArtifactParentWrapper.class, GSON);
+    } catch (IOException e) {
+      LOG.error("Exception reading artifacts named {} for namespace {} from the store.", artifactName, namespaceId, e);
+      responder.sendString(HttpResponseStatus.INTERNAL_SERVER_ERROR, "Error reading artifact metadata from the store.");
```

Comment:
I'm sure that this line of code is a copy of similar existing code in this class, but I was wondering about the usage of the term "store", and if that is the best term. What does it refer to? Not the Cask Market, which it might be confused with. Or is it?